### PR TITLE
Ap 3408 add los choice page

### PR DESCRIPTION
**Pull request by kmahern**
## What

[Link to story](https://dsdmoj.atlassian.net/browse/AP-3408)

Added emergency_level_of_service and substantive_level_of_service pages

Describe what you did and why.

## Checklist

Before you ask people to review this PR:

- Tests and rubocop should be passing: `bundle exec rake`
- Github should not be reporting conflicts; you should have recently run `git rebase main`.
- There should be no unnecessary whitespace changes. These make diffs harder to read and conflicts more likely.
- The PR description should say what you changed and why, with a link to the JIRA story.
- You should have looked at the diff against main and ensured that nothing unexpected is included in your changes.
- You should have checked that the commit messages say why the change was made.

### CODE REVIEW

File: spec/forms/proceedings/emergency_level_of_service_form_spec.rb [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    context "when the user doesn't answer the question" do
+      let(:emergency_level_of_service) { "" }
+
+      it "is invalid" do
+        expect(form).to be_invalid
+      end
+
+      #   it "generates the expected error message" do
```
Comment (colinbruce) [inline spec/forms/proceedings/emergency_level_of_service_form_spec.rb:46]: Can this be removed or implemented?

File: spec/forms/proceedings/substantive_level_of_service_form_spec.rb [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+    end
+
+    context "when the user doesn't answer the question" do
+      let(:substantive_level_of_service) { "" }
+
+      it "is invalid" do
+        expect(form).to be_invalid
+      end
+
+      #   it "generates the expected error message" do
```
Comment (colinbruce) [inline spec/forms/proceedings/substantive_level_of_service_form_spec.rb:46]: ditto?

File: spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+        let(:application) do
+          create(:legal_aid_application,
+                 :with_proceedings,
+                 :with_delegated_functions_on_proceedings,
+                 explicit_proceedings: %i[se013],
+                 df_options: { SE013: [10.days.ago, 10.days.ago] },
+                 substantive_application_deadline_on: 10.days.from_now)
+        end
+
+        it "displays a message that the level of service cannot be changed" do
```
Comment (colinbruce) [inline spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb:61]: I think this description needs updating?

File: spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+          expect(response.body).to include("For the emergency application, select the level of service")
+        end
+      end
+    end
+  end
+
+  describe "POST /providers/applications/:legal_aid_application_id/emergency_level_of_service/:proceeding_id" do
+    subject(:post_sd) { patch "/providers/applications/#{application_id}/emergency_level_of_service/#{proceeding_id}", params: }
```
Comment (colinbruce) [inline spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb:69]: Minor nitpick, can you rename the subject to `post_elos` so it matches the get variable above?

File: spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb [hunk 5]
```diff
@@ -0,0 +1,100 @@
...
+        let(:application) do
+          create(:legal_aid_application,
+                 :with_proceedings,
+                 :with_delegated_functions_on_proceedings,
+                 explicit_proceedings: %i[se013],
+                 df_options: { SE013: [10.days.ago, 10.days.ago] },
+                 substantive_application_deadline_on: 10.days.from_now)
+        end
+
+        it "displays a message that the level of service cannot be changed" do
```
Comment (colinbruce) [inline spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb:61]: ditto above, update the description

File: spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+          expect(response.body).to include("For the substantive application, select the level of service")
+        end
+      end
+    end
+  end
+
+  describe "POST /providers/applications/:legal_aid_application_id/substantive_level_of_service/:proceeding_id" do
+    subject(:post_sd) { patch "/providers/applications/#{application_id}/substantive_level_of_service/#{proceeding_id}", params: }
```
Comment (colinbruce) [inline spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb:69]: ditto above, update the subject name

**Review by colinbruce — COMMENTED** [review]
Looking great, but there are a couple of things that need looking at 👍

File: spec/forms/proceedings/emergency_level_of_service_form_spec.rb (hunk 1 above)
Comment (kmahern) [reply spec/forms/proceedings/emergency_level_of_service_form_spec.rb]: 👍 Have implemented this test

File: spec/forms/proceedings/substantive_level_of_service_form_spec.rb (hunk 2 above)
Comment (kmahern) [reply spec/forms/proceedings/substantive_level_of_service_form_spec.rb]: 👍 Have implemented this test

File: spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb (hunk 3 above)
Comment (kmahern) [reply spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb]: 👍 have updated

File: spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb (hunk 4 above)
Comment (kmahern) [reply spec/requests/providers/proceeding_loop/emergency_level_of_service_spec.rb]: 👍 have udpated

File: spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb (hunk 5 above)
Comment (kmahern) [reply spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb]: 👍

File: spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb (hunk 6 above)
Comment (kmahern) [reply spec/requests/providers/proceeding_loop/substantive_level_of_service_spec.rb]: 👍

**Review by colinbruce — APPROVED** [review]
Nice :+1:

**Pull request merged**